web/ui/src/app/lineage/graph/graph-builder.ts

```diff
@@ -34,13 +37,82 @@ export function lineageToGraph(lineage: IDataLineage,
         visibleOperations: IOperation[] = operationsByVisibility.true,
         hiddenOperations: IOperation[] = operationsByVisibility.false,
         hiddenOpIds: string[] = _.map(hiddenOperations, "mainProps.id"),
-        visibleNodes = visibleOperations.map(op => new RegularVisNode(op)),
+        visibleNodes = visibleOperations.map(op => new RegularVisNode(op, getLabel(op, expressionRenderService))),
         visibleEdges = createVisibleEdges(lineage, hiddenOpIds)
 
     return new VisModel(
         new vis.DataSet<VisNode>(visibleNodes),
         new vis.DataSet<VisEdge>(visibleEdges))
 }
+/**
+ * Get label from an operation
+ * 
+ * @param operation the operation where the label should be retrieved
+ * @param expressionRenderService service to render expressions
+ * 
+ * @returns the label from the operation
+ */
+export function getLabel(operation : IOperation, expressionRenderService : ExpressionRenderService){
+    let label = operation.mainProps.name + "\n"
+    switch(typeOfOperation(operation)){
+        case "Aggregate":
+            let aggregationExpressions = (<any> operation).aggregations
+            label += Object.keys(aggregationExpressions).join(",")
+            break
+        case "Join":
+            let joinExpression = (<IJoin> operation).condition
+            label += expressionRenderService.getText(joinExpression)
+            break
+        case "Filter":
+            let filterExpression = (<IFilter> operation).condition
+            label += expressionRenderService.getText(filterExpression)
+            break
+        case "Projection":
+            // TODO : Define what to display for the second label
+            break
+        case "Sort":
+            let sortExpressions = (<ISort> operation).orders
+            label += sortExpressions[0].direction + " " + expressionRenderService.getText(sortExpressions[0])
+            break
+        case "Read":
+            let readExpression = (<IRead> operation).sources
+            label += getFileName(readExpression[0].path)
+            break
+        case "Write":
+            label += (<IWrite> operation).path
+            break
+        default: 
+            label = operation.mainProps.name
+    }
+
+    return formatLabel(label)
+}
+
+/**
+ * Get the filename from a path
+ * 
+ * @param path string where the filename should be extracted
+ * 
+ * @returns the the filename of the path in parameters
+ */
+function getFileName(path: string): string{
+    return path.replace(/^.*[\\\/]/, '')
+}
+
+/**
+ * Takes the first maxSize characters and add ellipsis to it
+ * @param label the label to format
+ * @param maxSize the size of the final string
+ * 
+ * return the input label formated
+ */
+function formatLabel(label: string, maxSize: number = 50): string{
```

Comment:
This function is redundant. There is `truncate` method in LoDash that does exactly that. And it's also a little bit more accurate at placing "..." when the original string length is equal to the max length.
Example: `_.truncate(label, {length: maxSize})`

Reply:
corrected